Missing setjmp and longjmp in 32 bit libc for x86 (SDK 0.11.0)

It looks that symbols setjmp and longjmp are not defined in the 32bit version of libc.
As the result my project does not link properly (configured with CONFIG_NEWLIB_LIBC=y).

These symbols appear in the 64bit libc (as well as in all versions of ARM libraries).
Is it intentional?


```
hid@devel:/opt/zephyr-sdk$ nm -g x86_64-zephyr-elf/x86_64-zephyr-elf/lib/libc_nano.a | grep jmp
lib_a-setjmp.o:
0000000000000030 T longjmp
0000000000000000 T setjmp
hid@devel:/opt/zephyr-sdk$ nm -g x86_64-zephyr-elf/x86_64-zephyr-elf/lib/libc.a | grep jmp
lib_a-setjmp.o:
0000000000000030 T longjmp
0000000000000000 T setjmp
hid@devel:/opt/zephyr-sdk$ nm -g x86_64-zephyr-elf/x86_64-zephyr-elf/lib/32/libc.a | grep jmp
hid@devel:/opt/zephyr-sdk$ nm -g x86_64-zephyr-elf/x86_64-zephyr-elf/lib/32/libc_nano.a | grep jmp
hid@devel:/opt/zephyr-sdk$
```

@galak it works partially.

It works if I use offcial 0.11.1 sysroots because with the one from the file you mentioned I cannot configure my project.

```
-- Zephyr version: 2.1.99
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Selected BOARD qemu_x86
-- Found west: /home/hid/.local/bin/west (found suitable version "0.6.3", minimum required is "0.6.0")
CMake Error at /home/hid/prepilot/zephyr/cmake/host-tools.cmake:67 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  /home/hid/prepilot/zephyr/cmake/app/boilerplate.cmake:443 (include)
  CMakeLists.txt:2 (include)


CMake Error at /home/hid/prepilot/zephyr/cmake/extensions.cmake:1457 (message):
  Assertion failed: The detected dtc version is unsupported.

      The version was found to be 0                                   
      But the minimum supported version is 1.4.6                        
      See https://docs.zephyrproject.org/latest/getting_started/                     
      for how to use the SDK's dtc alongside a custom toolchain.
Call Stack (most recent call first):
  /home/hid/prepilot/zephyr/cmake/host-tools.cmake:69 (assert)
  /home/hid/prepilot/zephyr/cmake/app/boilerplate.cmake:443 (include)
  CMakeLists.txt:2 (include)
```
I also cannot see nano version of the newlib in menuconfig.

--- a/patches/newlib/3.1.0.20181231/0003-Support-building-for-32-bit-under-x86_64.patch
+++ b/patches/newlib/3.1.0.20181231/0003-Support-building-for-32-bit-under-x86_64.patch
@@ -1,4 +1,4 @@
-From 7dd18c12276cd4cb784dbebd9eb22eed45a49db6 Mon Sep 17 00:00:00 2001
+From dc48595340a292f044a0fddcab2d67a81650010f Mon Sep 17 00:00:00 2001
 From: Daniel Leung <daniel.leung@intel.com>
 Date: Wed, 8 May 2019 09:17:10 -0700
 Subject: [PATCH 3/3] Support building for 32-bit under x86_64
@@ -8,22 +8,25 @@ This is for multilib support under x86_64. When compiling for
 x86_64.
 
 Signed-off-by: Daniel Leung <daniel.leung@intel.com>
+Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
 ---
- newlib/configure.host | 12 +++++++++++-
- 1 file changed, 11 insertions(+), 1 deletion(-)
+ newlib/configure.host | 14 +++++++++++++-
+ 1 file changed, 13 insertions(+), 1 deletion(-)
 
 diff --git a/newlib/configure.host b/newlib/configure.host
-index 6c49cb7..0a40b1a 100644
+index 6c49cb750..37d2b9b2e 100644
 --- a/newlib/configure.host
 +++ b/newlib/configure.host
-@@ -333,7 +333,17 @@ case "${host_cpu}" in
+@@ -333,7 +333,19 @@ case "${host_cpu}" in
  	machine_dir=w65
  	;;
    x86_64)
 -	machine_dir=x86_64
 +	case "${CC}" in
 +	  *-m32*)
++	      libm_machine_dir=i386
 +	      machine_dir=i386
++	      mach_add_setjmp=true
 +	      ;;
 +	  *-mx32)
 +	      machine_dir=x86_64
@@ -36,5 +39,5 @@ index 6c49cb7..0a40b1a 100644
    xc16x*)
          machine_dir=xc16x
 -- 
-2.23.0
+2.24.1
 

